app/assets/javascripts/app/user_links.js

```diff
@@ -1,18 +1,44 @@
 App.addChild('UserLinksForm', {
+  
   el: '#links',
 
-  activate: function() {
-    this.nestedLinksSetup();
-    this.$el.on('cocoon:after-insert', function(e, insertedItem) {
-      that.nestedLinksSetup();
-    });
+  events: {
+    'click #add-link': 'addLink',
+    'click .remove_fields': 'removeLink'
   },
 
-  nestedLinksSetup: function() {
-    var that = this;
-    this.$('a.add-user-link').unbind('click');
-    this.$('a.add-user-link').bind('click', function(event) {
-      that.$('a.user-links.add_fields').trigger('click');
-    });
+  addLink: function(event){
+
+    event.preventDefault();
+
+    this.$('a.user-links.add_fields').trigger('click');
+
+  },
+
+  removeLink: function(event){
```

Comment:
Are you sure this is necessary? 
Take a look at these docs: https://github.com/nathanvda/cocoon#simpleform
I believe cocoon handles this automatically. @devton has some experience with this.